theano/tensor/nnet/abstract_conv.py

```diff
@@ -71,10 +70,8 @@ def get_conv_output_shape(image_shape, kernel_shape,
     """
     bsize, imshp = image_shape[0], image_shape[2:]
     nkern = kernel_shape[0]
-    if unshared is True:
-        kshp = kernel_shape[4:]
-    else:
-        kshp = kernel_shape[2:]
+
+    kshp = kernel_shape[-2:]
```

Comment:
It's nice to get rid of the `unshared` argument if you don't need it, but I'm not sure this is correct. I think it assumes that you do a 2D convolution?

Perhaps you can use the length of `image_shape` to compute the value of `convdim`, then use that instead of 2?